src/Runner/Runner.php

```diff
@@ -130,20 +130,28 @@ public function __construct(
     }
 
     /**
+     * @TODO consider to drop this method and make iterator parameter obligatory in constructor,
+     * more in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777/files#r1590447581
+     *
      * @param \Traversable<array-key, \SplFileInfo> $fileIterator
      */
     public function setFileIterator(iterable $fileIterator): void
     {
-        // @TODO consider to drop this method and make iterator parameter obligatory in constructor,
-        // more in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777/files#r1590447581
         $this->fileIterator = $fileIterator;
+
+        // Required only for main process (calculating workers count)
+        $this->fileCount = \count(iterator_to_array($fileIterator));
     }
 
     /**
      * @return _RunResult
      */
     public function fix(): array
     {
+        if (0 === $this->fileCount) {
```

Comment:
we do not need to bother fixing anything in anyway, if there are no files to be fixed.